beamer/tests/test_handlers.py

```diff
@@ -0,0 +1,72 @@
+from unittest.mock import MagicMock
+from beamer.agent import Config
+from beamer.chain import Context, claim_request
+from beamer.request import Request, Tracker
+from beamer.typing import ChainId, RequestId, TokenAmount
+from beamer.util import TokenMatchChecker
+
+
+class MockEth:
+    def __init__(self, chain_id):
+        self.chain_id = chain_id
+
+    def get_block(self, _block_identifier):  # pylint: disable=unused-argument, no-self-use
+        return {
+            "number": 42,
+            "timestamp": 457,
+        }
+
+
+class MockWeb3:
+    def __init__(self, chain_id):
+        self.eth = MockEth(chain_id=chain_id)
+
+
+def make_request(token):
+    return Request(
+        request_id=RequestId(1),
+        source_chain_id=ChainId(2),
+        target_chain_id=ChainId(4),
+        source_token_address=token.address,
+        target_token_address=token.address,
+        target_address=token.address,
+        amount=TokenAmount(123),
+        valid_until=456,
+    )
+
+
+def make_context(config, request_manager=None):
+    if request_manager is None:
+        request_manager = MagicMock()
+
+    checker = TokenMatchChecker([])
+
+    return Context(
+        Tracker(), Tracker(), request_manager, MagicMock(), checker, 5, config.account.address
+    )
+
+
+def test_skip_not_self_filled(token, config: Config):
+    r = make_request(token)
```

Comment:
Nitpick: I'd prefer if this was written in full: `request = ...`, like in other tests.